test/decl/subscript/static.swift

```diff
@@ -3,10 +3,34 @@
 // FIXME: Write proper tests (this file is mainly suitable for interactive testing)
 
 struct MyStruct {
-    static subscript(_ i: Int) -> String {
-        return "success"
-    }
+  static subscript(_ i: Int) -> String {
+    get { return "get \(i)" }
+    set { print("set \(i)") }
+  }
 }
 
 print(MyStruct.self[0])
-//print(MyStruct[0])
+print(MyStruct[0])
+
+MyStruct.self[1] = "zyzyx"
+MyStruct[2] = "asdfg"
+
+@dynamicMemberLookup
+class Dyn {
+  static subscript(dynamicMember name: String) -> String {
+    return "Dyn.\(name)"
+  }
+}
+
+print(Dyn.foo)
+print(Dyn.bar)
+
+class Base {
+  static subscript(_ i: Int) -> String { return "Base" }
+}
+class DerivedGood: Base {
+  override static subscript(_ i: Int) -> String { return "DerivedGood" }
+}
+
+print(DerivedGood[0])
```

Comment:
Can you have a test where you subscript a subclass but it *doesn't* override the subscript implementation?

Reply:
I suppose so.

(Actually, I just realized this test is wrong—the base class subscript should be implicitly `final`. I'll have to figure out what's happening there...)

Reply:
I've added some override tests to test/Interpreter/generic_subscript_static.swift, including one that tests a subscript that doesn't appear in the base class. I've also added some SILGen tests to make sure we handle class subscript overrides the same way as instance subscript overrides.